### PR TITLE
Workspace features Realm models

### LINKED ISSUE #397: Implement persisting WorkspaceFeatureCollection to the database
Follow-up of #214

### PR DESCRIPTION
**Pull request by simonrozsival**
Closes #397 

This PR includes only models, there is no repository in `Database`, as none is needed.

I also believe the models should not implement `IDatabaseSyncable`. The features collection is read-only and it should change only when new data is fetched from the API server.

### CODE REVIEW

File: Toggl.PrimeRadiant.Realm/Models/RealmWorkspaceFeatureCollection.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        public long WorkspaceId { get; set; }
+
+        public IList<RealmWorkspaceFeature> RealmWorkspaceFeatures { get; }
+
+        public IEnumerable<IDatabaseWorkspaceFeature> DatabaseFeatures
+            => RealmWorkspaceFeatures?.Cast<IDatabaseWorkspaceFeature>().ToList() ?? new List<IDatabaseWorkspaceFeature>();
+
+        public IEnumerable<IWorkspaceFeature> Features
+            => DatabaseFeatures.ToList<IWorkspaceFeature>();
```
Comment (simonrozsival) [inline Toggl.PrimeRadiant.Realm/Models/RealmWorkspaceFeatureCollection.cs]: The `ToList` calls are unnecessary and I will remove them, thanks!

The `IEnumerable` due to the implementation of the `IWorkspaceFeatureCollection` interface. The `IList` is required by Realm (https://realm.io/docs/xamarin/latest/#relationships).

**Comment by willsb** [comment]
I don't see the need to have the Workspace referencing this since, like Paul says, it'll be seldom used. And this feels a bit off, because the collection can't be consumed unless we cast the whole object to to a `RealmWorkspace`, since there's no reference to the `WorkspaceFeatureCollection` in the `IDatabaseWorkspace` interface

**Comment by simonrozsival** [comment]
The PR #421 must be closed first so it is possible to continue working on this PR, though I think that I will close this PR and do it from scratch, because this is very outdated and the idea of a merge with the current state of the repository is just scary.

**Comment by simonrozsival** [comment]
I am closing this - it is impossible to merge after the changes in the repositories. I will create this again from scratch.

**Comment by simonrozsival** [comment]
I am closing this - it is impossible to merge after the changes in the repositories. I will create this again from scratch.

**Pull request closed without merging**